lib/elixir/lib/macro.ex

```diff
@@ -315,6 +315,20 @@ defmodule Macro do
   def update_meta(other, _fun) do
     other
   end
+  @doc """
+  Generates AST nodes for a given number of required argument variables using
+  `Macro.var/2`.
+
+  ## Examples
+
+      iex> Macro.generate_arguments(2, __MODULE__)
+      [{:variable1, [], __MODULE__}, {:variable2, [], __MODULE__}]
+
+  """
+  def generate_arguments(number_of_args), do: generate_arguments(number_of_args, __MODULE__)
+  def generate_arguments(number_of_args, module) do 
```

Comment:
Also, maybe `context` is a better name for the second argument given it can be `nil` as well. 🤔